js/js.translator/src/org/jetbrains/kotlin/js/translate/test/JSTestGenerator.kt

```diff
@@ -72,62 +73,94 @@ class JSTestGenerator(val context: TranslationContext) {
         if (classDescriptor.modality === Modality.ABSTRACT || classDescriptor.isExpect) return
 
         val suiteFunction = JsFunction(context.scope(), JsBlock(), "suite function")
-
         val descriptors = classDescriptor.unsubstitutedMemberScope
                 .getContributedDescriptors(DescriptorKindFilter.FUNCTIONS, MemberScope.ALL_NAME_FILTER)
 
-        val beforeFunctions = descriptors.filterIsInstance<FunctionDescriptor>().filter { it.isBefore }
-        val afterFunctions = descriptors.filterIsInstance<FunctionDescriptor>().filter { it.isAfter }
+        val innerContext = context.contextWithScope(suiteFunction)
+        val classVal = classDescriptor.instance(innerContext)
+        val testInstanceName = JsName("instance${classDescriptor.name.toString()}")
+        val testInstanceRef = testInstanceName.makeRef()
+        val classVar = JsVars.JsVar(testInstanceName, classVal)
```

Comment:
This changes the way suites are instantiated. Previously a new instance was created for each test, now the suite instance is shared between tests.

I'm not sure this change is desirable (in my opinion it is not, as it reduces isolation). At the very least it requires motivation and discussion.